ui/components/OverviewComponent.js

```diff
@@ -0,0 +1,280 @@
+  },
+  Consul: {
+    name: "Consul",
+    img: "/static/img/consul.svg",
+    gradient: "linear-gradient(147deg, #F95089, #E05A41)",
+  },
+  "Network Service Mesh": {
+    name: "Network Service Mesh",
+    img: "/static/img/networkservicemesh.svg",
+    gradient: "linear-gradient(147deg, #FF9671, #EBC112)",
+  }
+};
+
+class OverviewComponent extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+
+      meshScan: {},
+      activeMeshScanNamespace: {},
+      meshScanNamespaces: {}
+    };
+  }
+
+  componentDidMount() {
+    this.fetchMeshScanData();
+  }
+
+  fetchMeshScanData = () => {
+    const self = this;
+    dataFetch(
+      "/api/mesh/scan",
+      {
+        credentials: "same-origin",
+        method: "GET",
+        credentials: "include",
+      },
+      (result) => {
+        if (result) {
+          // Extract all the unique namespaces in the mesh scan
+          const namespaces = {};
+          const activeNamespaces = {};
+          Object.keys(result).forEach(mesh => {
+            if (Array.isArray(result[mesh])) {
+              result[mesh].forEach(comp => {
+                if (comp.metadata) {
+                  if (namespaces[mesh]) namespaces[mesh].add(comp.metadata.namespace)
+                  else namespaces[mesh] = new Set([comp.metadata.namespace])
+                }
+              })
+              namespaces[mesh] = [...namespaces[mesh]]
+              activeNamespaces[mesh] = namespaces[mesh][0] || "";
+            }
+          })
+          self.setState({ meshScanNamespaces: namespaces, activeMeshScanNamespace: activeNamespaces });
+
+          // Check if Istio data is present in the scan
+          if (Array.isArray(result.Istio)) {
+            const istioData = result.Istio.map(comp => {
+              const compData = {
+                name: self.generateMeshScanPodName(
+                  comp.metadata.name, 
+                  comp.metadata.labels["pod-template-hash"],
+                  comp.metadata.generateName
+                ),
+                component: comp.metadata.labels?.app,
+                version: self.generateMeshScanVersion(comp.spec.containers?.[0]?.image),
+                namespace: comp.metadata.namespace
+              }
+              return compData;
+            })
+            self.setState(state => ({ meshScan: { ...state.meshScan, Istio: istioData } }));
+          }
+
+          // Check if Linkerd data is present in the scan
+          if (Array.isArray(result.Linkerd)) {
+            const linkerdData = result.Linkerd.map(comp => {
+              const compData = {
+                name: self.generateMeshScanPodName(
+                  comp.metadata.name, 
+                  comp.metadata.labels["pod-template-hash"],
+                  comp.metadata.generateName
+                ),
+                component: comp.metadata.labels["linkerd.io/control-plane-component"],
```

Comment:
We need a single, uniform way of identifying that a mesh is installed. 

A proof point for whether we hit the mark in MeshSync v2. // @kumarabd @vineethvanga18